code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -200,6 +221,17 @@
 	use_power(power_usage)
 	update_parents()
 
+/obj/machinery/atmospherics/components/unary/thermomachine/attackby(obj/item/assembly/signaler/anomaly/pyro, mob/living/user, params)
+	if(pyro_anomaly_core || !allows_cooling)
+		balloon_alert(user, "already an anomaly core!")
+		return
+	else
```

Comment:
This ``else`` is useless because you're early returning already

Reply:
Fixed